### ISSUE TITLE
PageTest.Control assertion failure

### ISSUE THREAD

**Issue by srh** [rethinkdb/rethinkdb#7044]
```
[----------] 15 tests from PageTest
[ RUN      ] PageTest.Control
Version: rethinkdb 2.4.1-97-gb79413-dirty (debug) (GCC 9.4.0)
error: Error in thread 1 in ./src/containers/intrusive_list.hpp at line 75:
Previous message may not have been written to the log file (logging module is not yet initialized).
error: Guarantee failed: [this->prev_ == this] non-empty intrusive list destroyed
Previous message may not have been written to the log file (logging module is not yet initialized).
error: Backtrace:
Previous message may not have been written to the log file (logging module is not yet initialized).
error: Fri Apr 15 14:46:40 2022
       
       1 [0x561359c010b7]: rethinkdb_backtrace(void**, int) at 0x561359c010b7 (./build/debug/rethinkdb-unittest)
       2 [0x56135a13aaf3]: backtrace_t::backtrace_t() at 0x56135a13aaf3 (./build/debug/rethinkdb-unittest)
       3 [0x56135a13b27c]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at 0x56135a13b27c (./build/debug/rethinkdb-unittest)
       4 [0x56135a13a9e9]: format_backtrace[abi:cxx11](bool) at 0x56135a13a9e9 (./build/debug/rethinkdb-unittest)
       5 [0x561359bfeb45]: report_fatal_error(char const*, int, char const*, ...) at 0x561359bfeb45 (./build/debug/rethinkdb-unittest)
       6 [0x561359bbfb52]: intrusive_list_t<linux_thread_message_t>::~intrusive_list_t() at 0x561359bbfb52 (./build/debug/rethinkdb-unittest)
       7 [0x561359bbfac6]: linux_message_hub_t::thread_queue_t::~thread_queue_t() at 0x561359bbfac6 (./build/debug/rethinkdb-unittest)
       8 [0x561359bbff4f]: std::enable_if<std::is_convertible<linux_message_hub_t::thread_queue_t (*) [], linux_message_hub_t::thread_queue_t (*) []>::value, void>::type std::default_delete<linux_message_hub_t::thread_queue_t []>::operator()<linux_message_hub_t::thread_queue_t>(linux_message_hub_t::thread_queue_t*) const at 0x561359bbff4f (./build/debug/rethinkdb-unittest)
       9 [0x561359bbfc5d]: std::unique_ptr<linux_message_hub_t::thread_queue_t [], std::default_delete<linux_message_hub_t::thread_queue_t []> >::~unique_ptr() at 0x561359bbfc5d (./build/debug/rethinkdb-unittest)
       10 [0x561359bbf186]: linux_message_hub_t::~linux_message_hub_t() at 0x561359bbf186 (./build/debug/rethinkdb-unittest)
       11 [0x561359bbd55c]: linux_thread_t::~linux_thread_t() at 0x561359bbd55c (./build/debug/rethinkdb-unittest)
       12 [0x561359bbbc70]: linux_thread_pool_t::start_thread(void*) at 0x561359bbbc70 (./build/debug/rethinkdb-unittest)
       13 [0x7f559da5e609]: /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7f559da5e609] at 0x7f559da5e609 (/lib/x86_64-linux-gnu/libpthread.so.0)
       14 [0x7f559d983163]: clone+0x43 at 0x7f559d983163 (/lib/x86_64-linux-gnu/libc.so.6)
Previous message may not have been written to the log file (logging module is not yet initialized).
error: Exiting.
Previous message may not have been written to the log file (logging module is not yet initialized).
Trace/breakpoint trap (core dumped)
```

**Comment by srh** [comment]
This doesn't occur if we run the test individually.  But it does happen consistently if we run

`./build/debug/rethinkdb-unittest --gtest_filter=BtreeMetadata.MetadataTest:PageTest.Control`

or if we try to run all the unit tests.

**Comment by srh** [comment]
I git bisected, and the first bad commit is bf8bde99cf74fa5bc0ea928e4e22a51811dc9459.

**Comment by srh** [comment]
The cause of the crash is that this static variable gets initialized with whatever the first value of n_threads is used.  It works fine in RethinkDB as a process, where we start the thread pool exactly once, and also only use one value of n_threads.  But it's not fine when running unit tests, under which the value of n_threads can be different.

This popped up as a result of the recent changes to remove MAX_THREADS.  For example, hard-coding 256 here would fix the crash.

```
perfmon_duration_sampler_t *pm_eventloop_singleton_t::get(int n_threads) {
    static perfmon_duration_sampler_t pm_eventloop(secs_to_ticks(1), false, n_threads);
```

**Comment by srh** [comment]
Fixed by #7045.  As mentioned, it only affected unit tests.
